src/recur.cpp

```diff
@@ -47,8 +47,28 @@
 #include <fstream>
 #include <iomanip>
 #include <iostream>
+#include <limits>
+#include <optional>
 #include <sstream>
 
+namespace {
+
+// Add a `time_t` delta to a Datetime, checking for and returning nullopt on integer overflow.
+std::optional<Datetime> checked_add(Datetime& base, time_t delta) {
```

Comment:
As a pure function, it is is a good candidate for a unit test.

Reply:
I wish there was a way to test private functions. I'll just make this one public to test it.